packages/hosted-credit-card-integration/src/HostedCreditCardPaymentMethod.tsx

```diff
@@ -0,0 +1,276 @@
+
+    const getHostedStoredCardValidationFieldset: (selectedInstrument: CardInstrument) => ReactNode =
+        useCallback(
+            (selectedInstrument) => {
+                const isInstrumentCardNumberRequired = selectedInstrument
+                    ? isInstrumentCardNumberRequiredProp(selectedInstrument)
+                    : false;
+                const isInstrumentCardCodeRequired = selectedInstrument
+                    ? isInstrumentCardCodeRequiredProp(selectedInstrument, method)
+                    : false;
+
+                return (
+                    <HostedCreditCardValidation
+                        cardCodeId={
+                            isInstrumentCardCodeRequired ? getHostedFieldId('ccCvv') : undefined
+                        }
+                        cardNumberId={
+                            isInstrumentCardNumberRequired
+                                ? getHostedFieldId('ccNumber')
+                                : undefined
+                        }
+                        focusedFieldType={focusedFieldType}
+                    />
+                );
+            },
+            [
+                focusedFieldType,
+                getHostedFieldId,
+                isInstrumentCardCodeRequiredProp,
+                isInstrumentCardNumberRequiredProp,
+                method,
+            ],
+        );
+
+    const initializePayment = checkoutService.initializePayment;
+
+    const initializeHostedCreditCardPayment: CreditCardPaymentMethodProps['initializePayment'] =
+        useCallback(
+            async (options, selectedInstrument) => {
+                return initializePayment({
+                    ...options,
+                    creditCard: {
+                        form: await getHostedFormOptions(selectedInstrument),
+                    },
+                });
+            },
+            [getHostedFormOptions, initializePayment],
+        );
+
+    const hostedStoredCardValidationSchema = getHostedInstrumentValidationSchema({ language });
+
+    const props = {
+        checkoutService,
+        checkoutState,
+        paymentForm,
+        language,
+        method,
+        onUnhandledError,
+    };
+
+    return (
+        <CreditCardPaymentMethodComponent
+            {...props}
+            cardFieldset={hostedFieldset}
+            cardValidationSchema={hostedValidationSchema}
+            deinitializePayment={checkoutService.deinitializePayment}
+            getHostedFormOptions={getHostedFormOptions}
+            getStoredCardValidationFieldset={getHostedStoredCardValidationFieldset}
+            initializePayment={initializeHostedCreditCardPayment}
+            storedCardValidationSchema={hostedStoredCardValidationSchema}
+        />
+    );
+};
+
+export default toResolvableComponent<PaymentMethodProps, PaymentMethodResolveId>(
+    HostedCreditCardPaymentMethod,
+    [
+        {
+            id: 'hosted-credit-card',
+            // type: 'PAYMENT_TYPE_API',
```

Comment:
If setting it as below, Google Pay wallet button will use this strategy at the payment step. Therefore, I changed it.
```
   id: 'credit-card',
   type: 'PAYMENT_TYPE_API',
```